rust/benches/ivf_pq.rs

```diff
@@ -61,11 +61,8 @@ async fn create_dataset(path: &std::path::Path, dim: usize, mode: WriteMode) {
     write_params.max_rows_per_file = num_rows as usize;
     write_params.max_rows_per_group = batch_size as usize;
     write_params.mode = mode;
-    let mut reader: Box<dyn RecordBatchReader> = Box::new(RecordBatchIterator::new(
-        batches.into_iter().map(Ok),
-        schema.clone(),
-    ));
-    Dataset::write(&mut reader, path.to_str().unwrap(), Some(write_params))
+    let reader = RecordBatchIterator::new(batches.into_iter().map(Ok), schema.clone());
+    Dataset::write(reader, path.to_str().unwrap(), Some(write_params))
```

Comment:
🤔  On one hand, we are eliminating a lot of boilerplate. On the other hand, it doesn't accept boxed RBR anymore, which is a bit annoying.

Reply:
We'll support boxed RBR again after https://github.com/apache/arrow-rs/pull/4475 fixed upstream and we pull that in.